packages/tinybased/src/lib/search/generateSearched.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { SearchParams, SearchResult } from '@lyrasearch/lyra';
+
+import { useCallback, useEffect, useState } from 'react';
+import { useValue } from 'tinybase/cjs/ui-react';
+import { ObjectToCellStringType, TableBuilder } from '../TableBuilder';
+import { TinyBased } from '../tinybased';
+import { OnlyStringKeys, TinyBaseSchema } from '../types';
+import { Searched } from './searched';
+
+const SEARCH_LAST_UPDATED_AT = (table: string | symbol | number) =>
+  `${String(table)}-last-updated-at`;
+
+/**
+ * Generates a type-safe Searched instance and its corresponding `useSearch` hook for a fully
+ * connected and reactive Full Text Search of an underlying `tinybased` dataset.
+ */
+export const generateSearched = async <
+  TBSchema extends TinyBaseSchema,
+  TIndexes extends OnlyStringKeys<TBSchema>
+>(
+  tinybased: TinyBased<TBSchema>,
+  indexes: Array<TIndexes>
+): Promise<{
+  searched: Searched<Pick<TBSchema, TIndexes>>;
+  useSearch: <K extends TIndexes>(
+    index: K,
+    params: SearchParams<ObjectToCellStringType<TBSchema[K]>>
+  ) => SearchResult<ObjectToCellStringType<TBSchema[K]>> | undefined;
+}> => {
+  // instantiate searched here
+  const searched = new Searched();
+
+  // for each table, addIndexedTable
+  const tables = new Map<TIndexes, TableBuilder<any, any>>();
+  indexes.forEach((tableKeyToIndex) => {
+    const tableBuilder = tinybased.tables.get(tableKeyToIndex);
+    if (!tableBuilder)
+      throw Error(Searched.ERRORS.TABLE_NOT_IN_SCHEMA_BUILDER(tableKeyToIndex));
+    tables.set(tableKeyToIndex, tableBuilder);
```

Comment:
What's the advantage of pushing them into the map here instead of just calling `addIndexedTable`?

Reply:
I was hoping for implicit type safety setting them all in one go outside of the loop. This library function already has liberal use of `any` though, so `addIndexedTable` in the loop and not being able to infer and use its type not the end of the world.